src/test/java/com/exasol/spark/common/FilterConverterTest.java

```diff
@@ -0,0 +1,239 @@
+package com.exasol.spark.common;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.jupiter.api.Assertions.assertAll;
+
+import java.math.BigDecimal;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import org.apache.spark.sql.sources.*;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import com.exasol.sql.expression.BooleanExpression;
+import com.exasol.sql.expression.ValueExpression;
+import com.exasol.sql.expression.rendering.ValueExpressionRenderer;
+import com.exasol.sql.rendering.StringRendererConfig;
+
+class FilterConverterTest {
+    private final FilterConverter filterConverter = new FilterConverter();
+
+    @Test
+    void testEmptyFilters() {
+        assertThat(render(), equalTo(""));
+    }
+
+    @Test
+    void testNullAndEmptyFilters() {
+        assertAll(() -> assertThat(filterConverter.convert(null), equalTo(Optional.empty())),
+                () -> assertThat(filterConverter.convert(new Filter[] {}), equalTo(Optional.empty())));
+    }
+
+    @Test
+    void testEmptyOptionalIfUnknownFilter() {
+        assertThat(filterConverter.convert(new Filter[] { new EqualNullSafe("c1", "nullValue") }),
+                equalTo(Optional.empty()));
+    }
+
+    private static final Stream<Arguments> unknownFilters() {
+        return Stream.of(//
+                Arguments.of(
+                        (Object) new Filter[] { new And(new EqualNullSafe("c1", "nullValue"), new EqualTo("c2", 1)) }), //
+                Arguments.of((Object) new Filter[] { new And(new LessThan("c2", 1), new EqualNullSafe("c1", "null")) }), //
+                Arguments.of((Object) new Filter[] { new Or(new EqualNullSafe("c1", "nv"), new EqualTo("c2", 1)) }), //
+                Arguments.of(
+                        (Object) new Filter[] { new Or(new GreaterThan("c2", 1), new EqualNullSafe("c1", "null")) }), //
+                Arguments.of((Object) new Filter[] { new Not(new EqualNullSafe("c1", "null")) }), //
+                Arguments.of((Object) new Filter[] { new LessThan("c2", "a"), new EqualNullSafe("c1", "nullValue") }) //
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("unknownFilters")
+    void testEmptyOptionalIfOneOfFiltersUnknown(final Filter[] filters) {
+        assertThat(filterConverter.convert(filters), equalTo(Optional.empty()));
+    }
+
+    @ParameterizedTest
+    @CsvSource({ //
+            "a,                      (\"c1\" = 'a')", //
+            "XYZ,                    (\"c1\" = 'XYZ')", //
+            "\u00d6,                 (\"c1\" = 'Ö')", //
+            "He said 'good morning', (\"c1\" = 'He said 'good morning'')" //
```

Comment:
Is this expectation correct?
```suggestion
            "He said 'good morning', (\"c1\" = 'He said ''good morning''')" //
```